build_barback/lib/src/transformer/transformer.dart

```diff
@@ -137,15 +138,13 @@ class _TransformAssetWriter implements AssetWriter {
   _TransformAssetWriter(this.transform);
 
   @override
-  Future writeAsBytes(build.AssetId id, List<int> bytes) {
-    transform.addOutput(barbackAssetFromBytes(id, bytes));
-    return new Future.value(null);
+  Future writeAsBytes(build.AssetId id, FutureOr<List<int>> bytes) async {
```

Comment:
I think it is important that this happens synchronously to make sure it gets waited on later?

You could verify by adding a test that returns synchronously from `build` after calling this method.